src/vos/vos_space.c

```diff
@@ -33,13 +32,8 @@ get_frag_overhead(daos_size_t tot_size, int media, bool small_pool)
 	if (media == DAOS_MEDIA_NVME)
 		return 0;
 
-	/* If caller specified the pool is small, do not enforce a range */
-	if (!small_pool) {
-		if (ovhd < min_sz)
-			ovhd = min_sz;
-		else if (ovhd > max_sz)
-			ovhd = max_sz;
-	}
+	if (ovhd > max_sz)
+		ovhd = max_sz;
 
```

Comment:
I think we could add some extra reservation for regular pool to mitigate the fragmentation issue.

if (tot_size >= (min_sz * 2) && !small_pool && ovhd < min_sz) {
    ovhd = min_sz;
}

Reply:
- [ ] Add some space for regular pool

Reply:
Fixed with commit https://github.com/daos-stack/daos/pull/13013/commits/270805402e811b5f153193363cf4304b7dcc26f0#diff-d43f9e7f3650645f86cda7b2c96d5e007c6b0567bdce7807f25da2274fa5b42dL22

Reply:
Fixed with commit https://github.com/daos-stack/daos/pull/13013/commits/270805402e811b5f153193363cf4304b7dcc26f0#diff-d43f9e7f3650645f86cda7b2c96d5e007c6b0567bdce7807f25da2274fa5b42dL22